### ISSUE TITLE
🐛 Panic `byte index 1 is out of bounds of ` when linting vue file

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#4583]
### Environment information

```block
CLI:
  Version:                      0.0.0 (self compiled d1ab6c54874d820be017d957c78808cbbd2333f7)
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```

<script>
}</script>
```
command
```
timeout -v 150 biome lint TEST___FILE.vue
```
cause this
```
/opt/BROKEN_FILES_DIR/transfer_IDX_3_RAND_566755088255087822711668_minimized_272.vue:3:1 parse ━━━━━━━━━━

  × Expected a statement but instead found '}'.
  
    2 │ <script>
  > 3 │ }</script>
      │ ^
  
  i Expected a statement here.
  
  > 1 │ 
      │ Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_diagnostics/src/display/frame.rs:240:57
Thread Name: main
Message: byte index 1 is out of bounds of ``


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

[compressed.zip](https://github.com/user-attachments/files/17809602/compressed.zip)

### Expected result

No crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
While not exactly a duplicate, I think the root cause is the same as https://github.com/biomejs/biome/issues/4593#issuecomment-2583145999

I'll close this to avoid getting too many issues related to this.
